.github/workflows/auto-update-gemfiles.yml

```
@@ -0,0 +1,72 @@
+name: Auto-update Gemfiles
+
+# This action cannot be skipped altogether because it is a mandatory status check.
+# Instead we conditionally skip it at job level, instead of workflow level.
+on:
+  pull_request:
+    paths:
+      - 'tasks/edge.rake'
+      - 'Gemfile'
+      - 'Gemfile.lock'
+      - 'appraisal/**'
+      - 'gemfiles/**'
+  workflow_dispatch:
+    
+
+# Ensure obsolete job is cancelled if another commit is pushed to the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  update-gemfiles:
+    name: Auto-update Gemfiles
+    runs-on: ubuntu-22.04
+    permissions:
+        pull-requests: write
+    strategy:
+        matrix:
+            ruby: ['2.5', '2.6', '2.7', '3.0', '3.1', '3.2', '3.3', '3.4']
```

Minor: Because there's like ~20 places that need to be updated once a year when a new Ruby version drops, we're trying to make sure they all get marked with this comment so we can find them easily and don't need to do a trial-and-error or weird searches to find them :)

```suggestion
        matrix:
            # ADD NEW RUBIES HERE
            ruby: ['2.5', '2.6', '2.7', '3.0', '3.1', '3.2', '3.3', '3.4']
```

---

Thank you for pointing this out! I added this :)